lib/surface/live_view.ex

```diff
@@ -63,9 +63,10 @@ defmodule Surface.LiveView do
 
   defp quoted_mount(env) do
     defaults =
-      for %{name: name, opts: opts} <- Module.get_attribute(env.module, :data) do
```

Comment:
I guess you could keep `for` and just add the condition as second argument in order to avoid `filter` and `map`:

```Elixir
defaults =
  for %{name: name, opts: opts} <- Module.get_attribute(env.module, :data), Keyword.has_key?(opts, :default) do
    {name, Keyword.get(opts, :default)}
  end
  |> Macro.escape()
```

BTW, here is a nice blog post about for comprehensions: https://www.pompecki.com/post/for-comprehensions/